src/WebJobs.Extensions.DurableTask/Options/TraceOptions.cs

```diff
@@ -64,6 +64,13 @@ public class TraceOptions
         /// </summary>
         public string DistributedTracingProtocol { get; set; } = "HttpCorrelationProtocol";
 
+        /// <summary>
+        /// Gets or sets a flag indicating whether to enable the new version of distributed tracing
+        /// that is built using OpenTelemetry.
+        /// The default value is false.
+        /// </summary>
+        public bool NewDistributedTracingEnabled { get; set; } = false;
+
```

Comment:
I left this name for now, but I want to change it before merging since it's not very self-explanatory. The current setting `DistributedTracingEnabled` is used in various different places for the previous implementation so I was hesitant to use that setting. I'm open to suggestions on how to handle this.

Reply:
Are we looking to completely replace the old implementation of Distributed Tracing once this goes into public preview? If so - I'd see the benefit in re-using the name.

Reply:
The plan right now is to keep the old implementation until GA.

Reply:
I like this idea! Existing apps that enabled distributed tracing wouldn't break and the OTel compatible implementation is opt-in.

Reply:
> We might as well keep the old stuff around but give a way to control what mode the customer uses.

@jviau just to clarify, are you suggesting we have this extra setting just for preview and then remove it for GA?

Reply:
@cgillum I don't think we should remove the existing distributed tracing even after GA. At least not right away. We should give a deprecation window so customers that already use it have a chance to migrate over. We may want to track this work, but we should probably give an in-portal alert for customers to switch over to new distributed tracing, and give a timeline that it will be removed at a certain date (we can decide on that date).

Reply:
Okay, but we probably need @lilyjma to weigh in on this decision.

Reply:
Thanks for looping me in. 

I've discussed with Varshi. Since replacing the old implementation would break current users, Jacob's suggestion sounds reasonable. Also, because the old implementation has been in preview for >1 year, we'll need to treat it as GA when retiring it, _assuming_ that it has >1k customers. Per Azure Retirement process, we'll need to give 18 month notice to retire the old implementation when there's a GA replacement (we could also replace with preview but that would require CTO approval). If we have <1k customers on the old implementation, the process would be easier. So the first step is to get the number of customers using the old implementation.